lib/src/rules/invariant_booleans.dart

```diff
@@ -172,6 +174,7 @@ class _InvariantBooleansVisitor extends ConditionScopeVisitor {
     final elements = _getElementsInExpression(node);
     Iterable<Expression> conjunctions = getTrueExpressions(elements).toSet();
     Iterable<Expression> negations = getFalseExpressions(elements).toSet();
-    return TestedExpressions(node, conjunctions, negations);
+    return TestedExpressions(
+        node, LinkedHashSet.from(conjunctions), LinkedHashSet.from(negations));
```

Comment:
Given that we're using `toSet` above (lines 175 and 176), do we really need to call these constructors? Could we just change the types of these variables to `Set<Expression>`?

Reply:
Great catch.  Thanks!